pyk/src/pyk/kcfg.py

```diff
@@ -115,9 +116,9 @@ def to_dict(self) -> Dict[str, Any]:
 
         def pretty_print(self, kprint: KPrint) -> List[str]:
             return [
-                'constraint: ' + kprint.pretty_print(self.constraint),
+                'constraint: ' + kprint.pretty_print(ml_pred_to_bool(self.constraint)),
                 'subst:',
-                *add_indent('  ', kprint.pretty_print(substToMlPred(self.subst)).split('\n')),
+                *add_indent('  ', kprint.pretty_print(substToMap(minimizeSubst(self.subst))).split('\n')),
```

Comment:
I think this should have the same effect:

```suggestion
                *add_indent('  ', kprint.pretty_print(dict(self.subst.minimize())).split('\n')),
```

If so, `substToMap` and `minimizeSubst` should be removed.